### PR TITLE
Add Information Protection APIs

### PR DESCRIPTION
**Pull request by ranisha2**
Previously reviewed PR:
https://github.com/Azure/azure-rest-api-specs/pull/3119

Add Information Protection APIs:
-GET Informaton Protection Policy
-PUT Information Protetion Policy
-LIST Information Protection Policies

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by hovsepm** [comment]
@ranisha2 please pull from upstream and resolve merge conflicts.

File: specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/security.json [hunk 1]
```diff
@@ -920,6 +920,116 @@
...
+                    },
+                    "default": {
+                        "description": "Error response describing why the operation failed.",
+                        "schema": {
+                            "$ref": "#/definitions/CloudError"
+                        }
+                    }
+                }
+            },
+            "put": {
```
Comment (hovsepm) [inline specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/security.json]: >put [](start = 13, length = 3)

no delete operation?

File: specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/security.json [hunk 2]
```diff
@@ -920,6 +920,116 @@
...
+        "/{scope}/providers/Microsoft.Security/informationProtectionPolicies": {
+            "get": {
+                "x-ms-examples": {
+                    "Get information protection policies": {
+                        "$ref": "./examples/InformationProtectionPolicies/ListInformationProtectionPolicies_example.json"
+                    }
+                },
+                "tags": ["InformationProtectionPolicies"],
+                "description": "Information protection policies of a specific management group.",
+                "operationId": "InformationProtection_List",
```
Comment (hovsepm) [inline specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/security.json]: >InformationProtection_List [](start = 32, length = 26)

should be `InformationProtectionPolicies_List`

**Comment by hovsepm** [comment]
@ranisha2 
Please fix model validation errors
```
Validating "examples" and "x-ms-examples" in  specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/security.json:
 error: 
operationId: InformationProtectionPolicies_Get
scenario: Get the customized information protection policy for a management group
source: request
responseCode: ALL
severity: 0
errorCode: REQUIRED_PARAMETER_EXAMPLE_NOT_FOUND
errorDetails:
  code: REQUIRED_PARAMETER_EXAMPLE_NOT_FOUND
  id: OAV105
  message: >-
    In operation "InformationProtectionPolicies_Get", parameter
    informationProtectionPolicyName is required in the swagger spec but is not
    present in the provided example parameter values.
  innerErrors: !<tag:yaml.org,2002:js/undefined> ''
 error: 
operationId: InformationProtectionPolicies_Get
scenario: Get the effective information protection policy for a management group
source: request
responseCode: ALL
severity: 0
errorCode: REQUIRED_PARAMETER_EXAMPLE_NOT_FOUND
errorDetails:
  code: REQUIRED_PARAMETER_EXAMPLE_NOT_FOUND
  id: OAV105
  message: >-
    In operation "InformationProtectionPolicies_Get", parameter
    informationProtectionPolicyName is required in the swagger spec but is not
    present in the provided example parameter values.
  innerErrors: !<tag:yaml.org,2002:js/undefined> ''
 error: 
operationId: InformationProtectionPolicies_CreateOrUpdate
scenario: Create or update an information protection policy for a management group
source: request
responseCode: ALL
severity: 0
errorCode: ENUM_MISMATCH
errorDetails:
  code: ENUM_MISMATCH
  params:
    - default
  message: 'No enum match for: default'
  path: >-
    /{scope}/providers/Microsoft.Security/informationProtectionPolicies/{informationProtectionPolicyName}/informationProtectionPolicyName
  description: Name of the information protection policy.
 error: 
operationId: InformationProtectionPolicies_CreateOrUpdate
scenario: Create or update an information protection policy for a management group
source: response
responseCode: '200'
severity: 0
errorCode: INVALID_TYPE
errorDetails:
  code: INVALID_TYPE
  params:
    - object
    - undefined
  message: Expected type object but found type undefined
  path: ''
  title: /definitions/InformationProtectionPolicy
  description: Information protection policy.
  position:
    line: 1562
    column: 40
  url: >-
    specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/security.json
 error: 
operationId: InformationProtectionPolicies_CreateOrUpdate
scenario: Create or update an information protection policy for a management group
source: response
responseCode: '201'
severity: 0
errorCode: RESPONSE_STATUS_CODE_NOT_IN_EXAMPLE
errorDetails:
  code: RESPONSE_STATUS_CODE_NOT_IN_EXAMPLE
  id: OAV111
  message: >-
    Following response status codes "201" for operation
    "InformationProtectionPolicies_CreateOrUpdate" were present in the swagger
    spec, however they were not present in x-ms-examples. Please provide them.
  innerErrors: !<tag:yaml.org,2002:js/undefined> ''
  level: error
```

**Comment by hovsepm** [comment]
@ranisha2 please also take a look to my comment above "should be InformationProtectionPolicies_List"

File: specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/security.json (hunk 1 above)
Comment (ranisha2) [reply specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/security.json]: There's no delete operation, user can override the policy with an empty set of labels/information types.

File: specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/security.json (hunk 2 above)
Comment (ranisha2) [reply specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/security.json]: Fixed. Thanks

**Comment by hovsepm** [comment]
@MikhailTryakhov, @owolabi2, @yangyuan could you review and sign off?

**Review by hovsepm — APPROVED** [review]

**Comment by hovsepm** [comment]
Adding @dsgouda . Seems that you have way much more context on this PR :) Please take a look.

**Comment by dsgouda** [comment]
@ranisha2 Please address @hovsepm 's comments and ARM feedback, will post comments from my end in the meantime

**Comment by ranisha2** [comment]
@dsgouda I already addressed these comments. Also, please note that this was already reviewed here (I addressed the comments there too): https://github.com/Azure/azure-rest-api-specs/pull/3119
(This one has some minor fixes and changes)

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**